python/opendataloader-pdf-mcp/src/opendataloader_pdf_mcp/mineru.py

```
@@ -0,0 +1,55 @@
+@dataclass
+class MinerUResult:
+    markdown: str
+    json_str: str
+    exit_code: int
+    stderr: str
+
+
+class MinerURunner:
+    def run(self, pdf_path: Path, output_dir: Path) -> MinerUResult:
+        if shutil.which("mineru") is None:
+            raise MinerUError("mineru not found in PATH")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)